src/Common/PODArray.h

```diff
@@ -104,8 +105,15 @@ class PODArrayBase : private boost::noncopyable, private TAllocator    /// empty
     char * c_end_of_storage = null;    /// Does not include pad_right.
 
     /// The amount of memory occupied by the num_elements of the elements.
-    static size_t byte_size(size_t num_elements) { return num_elements * ELEMENT_SIZE; }
-
+    static size_t byte_size(size_t num_elements) 
+    {
+        size_t test;
+        if (__builtin_mul_overflow(num_elements,ELEMENT_SIZE, &test))
+           throw Exception("Amount of memory requested to allocate is more than allowed", ErrorCodes::CANNOT_ALLOCATE_MEMORY);
```

Comment:
```suggestion
            throw Exception("Amount of memory requested to allocate is more than allowed", ErrorCodes::CANNOT_ALLOCATE_MEMORY);
```

Reply:
Hi @alexey-milovidov, Thanks, updated the code now.